tests/rest/client/test_rooms.py

```diff
@@ -1116,6 +1116,320 @@ def test_rooms_messages_sent(self) -> None:
+        """
+        With a server configured to modify the room-level defaults,
+        Create a room, providing different overrides.
+        Confirm that the room's power levels reflect both overrides, and
+        choose the room overrides where they clash.
+        """
+
+        room_id = self.helper.create_room_as(
+            self.user_id,
+            extra_content={
+                "power_level_content_override": {"events": {"room.event": 0}}
+            },
+        )
+
+        self.assertEqual(
+            {
+                "ban": 13,  # From server config override
+                "events": {
+                    "room.event": 0,  # From room override
+                },
+                "events_default": 0,
+                "historical": 100,
+                "invite": 50,
+                "kick": 50,
+                "redact": 50,
+                "state_default": 50,
+                "users": {"@sid1:red": 100},
+                "users_default": 0,
+            },
+            self.power_levels(room_id),
+        )
+
+
+class RoomPowerLevelOverridesInPracticeTestCase(RoomBase):
+    """Tests that we can really do various otherwise-prohibited actions
+    based on overriding the power levels in config."""
+
+    user_id = "@sid1:red"
+
+    def test_creator_can_post_state_event(self) -> None:
+        # Given I am the creator of a room
+        room_id = self.helper.create_room_as(self.user_id)
+
+        # When I send a state event
+        path = "/rooms/{room_id}/state/custom.event/my_state_key".format(
+            room_id=urlparse.quote(room_id),
+        )
+        channel = self.make_request("PUT", path, "{}")
+
+        # Then I am allowed
+        self.assertEqual(200, channel.code, msg=channel.result["body"])
+
+    def test_normal_user_can_not_post_state_event(self) -> None:
+        # Given I am a normal member of a room
+        room_id = self.helper.create_room_as("@some_other_guy:red")
+        self.helper.join(room=room_id, user=self.user_id)
+
+        # When I send a state event
+        path = "/rooms/{room_id}/state/custom.event/my_state_key".format(
+            room_id=urlparse.quote(room_id),
+        )
+        channel = self.make_request("PUT", path, "{}")
+
+        # Then I am not allowed because state events require PL>=50
+        self.assertEqual(403, channel.code, msg=channel.result["body"])
+        self.assertEqual(
+            "You don't have permission to post that to the room. "
+            + "user_level (0) < send_level (50)",
+            channel.json_body["error"],
+        )
+
+    @unittest.override_config(
+        {
+            "default_power_level_content_override": {
+                "public_chat": {"events": {"custom.event": 0}},
+            }
+        },
+    )
+    def test_with_config_override_normal_user_can_post_state_event(self) -> None:
+        # Given the server has config allowing normal users to post my event type
+        # And I am a normal member of a room
```

Comment:
nit: I'm finding the lack of punctuation between (partial) sentences here a bit odd to read.

I would normally write:

```suggestion
        # Given the server has config allowing normal users to post my event type,
        # and I am a normal member of a room
```

Reply:
I'm vaguely using [BDD/Cucumber style](https://behave.readthedocs.io/en/stable/tutorial.html) but I'll fix it.

Reply:
Fixed in 08b00e1b87e

Reply:
Oh, this is interesting and something I haven't come across before. Though I still think it's a little confusing given that the convention doesn't (yet) exist elsewhere in the codebase.

The lack of punctuation just confused me. Once could also write it as:

```
Given: the server has config allowing normal users to post my event type
And: I am a normal member of a room
Then: ...
```

and probably get away without it looking like odd grammar.

Reply:
Are you happy with it as it is now? (I am)